polyfill/lib/duration.mjs

```diff
@@ -291,15 +291,15 @@ export class Duration {
       microseconds,
       nanoseconds
     );
-    options = ES.GetOptionsObject(options);
-    let smallestUnit = ES.ToSmallestTemporalUnit(options, undefined);
+    roundTo = ES.Type(roundTo) === 'String' ? { smallestUnit: roundTo } : ES.GetOptionsObject(roundTo);
```

Comment:
I overlooked this before, but it just occurred to me that this is not exactly according to the specification, which says to do OrdinaryObjectCreate(*null*) followed by CreateDataPropertyOrThrow. So, `roundTo` should be an object with null prototype instead of Object.prototype. I think this is observable, because if it's an object with Object.prototype, the lookup of the `largestUnit` property would fetch `Object.prototype.largestUnit` if someone (with a terrible sense of humor) were to define that.

We should probably add a test262 test for that, but certainly no need to wait on that. I think it's sufficient to fix this code and the similar instances below to create a null-prototype object.

Reply:
OK will do. Is it only property bag parameters which should have this behavior?  Or should all observable object creation do it?  I ask because when I grep for `return {` in the polyfill source code I see about 100 matches. From a spot-check, at least some of those are observable (e.g. objects that are returned back to the user). Obviously we're not gonna fix those unrelated things in this PR, but was wondering if I should open an issue to scrub all object creation in the polyfill.

Reply:
I think we had a discussion about this with Kevin Gibbons in an issue. I can probably find it later if you want to read it. In cases where we just return the object to the user, like in the getISOFields methods, that's no problem. If they want to make it do weird magical property lookups in their own code at that point, that has nothing to do with Temporal. But in cases where the subsequent property lookup is done inside Temporal, we should have a null-prototype object.

You may be right that there are other existing cases where options objects are not conformant to the spec text.

Also somewhat related to https://github.com/tc39/proposal-temporal/issues/1685

Reply:
I just pushed a fix for this PR so we should be good to merge once the Test262 changes land and we can update this PR to use them. 

> But in cases where the subsequent property lookup is done inside Temporal, we should have a null-prototype object.

Got it. Looks like #1685 has things under control. Thanks for explaining.